packages/cli/src/commands/__tests__/generate-set-option.ts

```diff
@@ -0,0 +1,179 @@
+import {
+  parseSetOptions,
+  createCLIVariablesLayer
+} from "../generate-set-option";
+
+describe("Set Argument parser should return correct value", () => {
+  test("when one string argument is passed", () => {
+    expect(parseSetOptions("a=b")).toEqual([{ key: "a", value: "b" }]);
+  });
+
+  test("when several string arguments are passed", () => {
+    expect(parseSetOptions(["a=b", "c=d", "e=f"])).toEqual([
+      { key: "a", value: "b" },
+      { key: "c", value: "d" },
+      { key: "e", value: "f" }
+    ]);
+  });
+
+  test("when object with single value is passed", () => {
+    expect(parseSetOptions({ component: "a=b" })).toEqual([
+      { componentName: "component", key: "a", value: "b" }
+    ]);
+  });
+
+  test("when object with multiple values are passed", () => {
+    expect(parseSetOptions({ component: ["a=b", "c=d"] })).toEqual([
+      { componentName: "component", key: "a", value: "b" },
+      { componentName: "component", key: "c", value: "d" }
+    ]);
+  });
+
+  test("when object with multiple keys are passed", () => {
+    expect(parseSetOptions({ component: "a=b", component2: "c=d" })).toEqual([
+      { componentName: "component", key: "a", value: "b" },
+      { componentName: "component2", key: "c", value: "d" }
+    ]);
+  });
+
+  test("when object with multiple keys and values are passed", () => {
+    expect(
+      parseSetOptions({
+        component: ["k1=v1", "k2=v2"],
+        component2: ["k3=v3", "k4=v4"]
+      })
+    ).toEqual([
+      { componentName: "component", key: "k1", value: "v1" },
+      { componentName: "component", key: "k2", value: "v2" },
+      { componentName: "component2", key: "k3", value: "v3" },
+      { componentName: "component2", key: "k4", value: "v4" }
+    ]);
+  });
+
+  test("when object with single value and several string arguments are passed", () => {
+    expect(parseSetOptions([{ component: "a=b" }, "c=d", "e=f"])).toEqual([
+      { componentName: "component", key: "a", value: "b" },
+      { key: "c", value: "d" },
+      { key: "e", value: "f" }
+    ]);
+  });
+
+  test("when a number is passed as a value", () => {
+    expect(parseSetOptions("a=1")).toEqual([{ key: "a", value: 1 }]);
+  });
+
+  test("when an array is passed as a value", () => {
+    expect(parseSetOptions("a=[1,2,3]")).toEqual([
+      { key: "a", value: [1, 2, 3] }
+    ]);
+  });
+
+  test("when an object is passed as a value", () => {
+    expect(parseSetOptions('a={"b": 1}')).toEqual([
+      { key: "a", value: { b: 1 } }
+    ]);
+  });
```

Comment:
You should add tests of parsing JSON path key.

Reply:
I've added a test on parsing complex string with a bunch of "equal" chars in it.